tests/types/npm-test.json

```diff
@@ -254,6 +254,316 @@
+    },
+    {
+      "description": "Negative Test: An npm purl with a unicode character (emoji) in the name should fail parsing",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/foo%F0%9F%98%83bar@1.0.0",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse because the name contains characters not allowed by npm naming rules"
+    },
+    {
+      "description": "Negative Test: An npm purl with a unicode character (emoji) in the scope should fail parsing",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/%40foo%F0%9F%98%83/bar@1.0.0",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse because the scope contains characters not allowed by npm naming rules"
+    },
+    {
+      "description": "Positive Test: An npm purl with a subpath containing unicode characters (emoji) should roundtrip correctly",
+      "test_group": "advanced",
+      "test_type": "roundtrip",
+      "input": "pkg:npm/foo@1.0.0#src/%F0%9F%98%83/file.js",
+      "expected_output": "pkg:npm/foo@1.0.0#src/%F0%9F%98%83/file.js",
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Positive Test: An npm purl with a subpath containing unicode characters (emoji) should parse correctly",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/foo@1.0.0#src/%F0%9F%98%83/file.js",
+      "expected_output": {
+        "type": "npm",
+        "namespace": null,
+        "name": "foo",
+        "version": "1.0.0",
+        "qualifiers": null,
+        "subpath": "src/😃/file.js"
+      },
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Positive Test: An npm purl with a qualifier value containing unicode characters (emoji) should roundtrip correctly",
+      "test_group": "advanced",
+      "test_type": "roundtrip",
+      "input": "pkg:npm/foo@1.0.0?vcs_url=https://a.com/b/c%F0%9F%98%83d.git",
+      "expected_output": "pkg:npm/foo@1.0.0?vcs_url=https://a.com/b/c%F0%9F%98%83d.git",
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Positive Test: An npm purl with a qualifier value containing unicode characters (emoji) should parse correctly",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/foo@1.0.0?vcs_url=https://a.com/b/c%F0%9F%98%83d.git",
+      "expected_output": {
+        "type": "npm",
+        "namespace": null,
+        "name": "foo",
+        "version": "1.0.0",
+        "qualifiers": {
+          "vcs_url": "https://a.com/b/c😃d.git"
+        },
+        "subpath": null
+      },
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
```

Comment:
Do these tests have anything to do with NPM? They seem like they should be basic tests for the core spec.